plugin/src/main/kotlin/org/neotech/plugin/rootcoverage/CoverageReportTask.kt

```diff
@@ -0,0 +1,56 @@
+package org.neotech.plugin.rootcoverage
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.Project
+import org.gradle.testing.jacoco.tasks.JacocoReport
+import org.neotech.plugin.rootcoverage.utilities.outputReport
+
+/**
+ * Encapsulates configuration and logic to create a CoverageReportTask.
+ */
+class CoverageReportTask private constructor() : DefaultTask() {
```

Comment:
I see what you are trying to accomplish here, however I don't think this is the right way.

You created a new Gradle task `CoverageReportTask` which inherits from `DefaultTask`, however as of now this task is not used, so why is this class even inheriting from `DefaultTask`?

It seems like the only reason for this class to exist is to have a companion object with a "static" method in it that creates a new `JacocoReport` task. Don't get me wrong, but this feels a bit old fashioned Java style to me.

I like the idea of extracting the `createSubProjectCoverageTask` to a different file to make the main plugin class a bit less overcrowded, so I would suggest the following:

1. Rename this file to: `JacocoTaskFactory.kt`
2. Remove the class and companion object
3. Make the `create` function an extension function on `Project`, put it in the file as a fully stand-alone function (basically similar to how you did `Project.outputReport()`)
4. Rename the create function to `createJacocoReportTask()`

You basically end up with something like this:

`fun Project.createJacocoReportTask(rootProjectExtension: RootCoveragePluginExtension): JacocoReport`


_Note: In the past this plugin would actually create specific tasks (DefaultTask) besides just JacocoReport tasks, however this led to some problems. These tasks first partially handled doing configuration, which should normally happen in the configuration phase of Gradle and not during the execution phase. Once that was fixed the tasks where basically just there to configure the dependencies between tasks, making them pretty much useless. Hence they got removed in favor of configuring just `JacocoReport` tasks and making dependencies between those tasks and the Android Gradle Plugin test tasks directly.`_

Reply:
:heavy_check_mark: Done. I agree that there not much of a reason to wrap the function in a class. I disagree about the function name - it no longer tells the call side what is constructed - which is a `coverageReport` task. One would need to read the implementation to find out which should not be needed IMHO.